active/gridsyntax/GridSyntaxSpec2.md

```diff
@@ -0,0 +1,368 @@
+</Grid>
+```
+
+You can now accomplish the same thing in a single tag:
+
+```xml
+<Grid ColumnDefinitions="Auto, *">
+```
+
+Before this change, if you write that shorter Xaml markup, the  Xaml loader would try to convert the string `Auto, *` into a [ColumnDefinitionCollection](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ColumnDefinitionCollection) object, as that's the type of the [Grid.ColumnDefinitions](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Grid.ColumnDefinitions) property. That fails because there is no such conversion available. (And it wouldn't help if there was, because the property is read-only.)
+
+But the Xaml syntax rules now support this scenario by recognizing that the target property is a collection type, and that there's no conversion from string available, and then by treating the attribute value as a comma-separated list of collection items. The items of the collection are of type [ColumnDefinition](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ColumnDefinition), so a check is made to see if strings can be converted to ColumnDefinition instances. Since they can, the Xaml loader creates those instances and adds them to the collection.
+
+| Do we have documentation that explains what exactly a "collection" is? Can't remember if it's just IVector, or if it has more rules like the C# initialization syntax.
+
+## Role of the [ContentProperty] attribute
+
+In order for a ColumnDefinition (or RowDefinition) to be created from a string there is a second new Xaml syntax rule: if a type can't be created from string, but the type's [[ContentProperty](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Markup.ContentPropertyAttribute)] property can, the type is created and the property set.
+
+About this "content property", it's not literally a property named "content" (although it can be and [is sometimes](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ContentControl.Content)). This is essentially the default property of a class, and is marked with the [ContentProperty] attribute.
+
+For example the [SolidColorBrush](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush) is defined roughly like this (in C# terms):
+
+```cs
+[ContentProperty(Name = "Color")]
+public sealed class SolidColorBrush : Brush
+{
+    ...
+    public Color Color { get; set; }
+}
+```
+
+Because [SolidColorBrush.Color](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush.Color) is marked as the [ContentProperty] (and because a Color can be created from a string), you've always been able to write this short-hand Xaml:
+
+```xml
+<SolidColorBrush>Red</SolidColorBrush>
+```
+
+If it weren't for the [ContentProperty] on SolidColorBrush, you'd have to use the equivalent full syntax (and that's still valid to use if you prefer):
+
+```xml
+<SolidColorBrush>
+    <SolidColorBrush.Color>
+        Red
+    </SolidColorBrush.Color>
+</SolidColorBrush>
+```
+
```

Comment:
should this be in its own spec or doc since it's used by the grid syntax but not specific to it?

Reply:
This whole Discussion section is written like a blog post. This section on the CPA is a backgrounder on how it works, in order to understand what follows in the spec.